components/automate-ui/src/app/components/notification/notification.component.spec.ts

```diff
@@ -0,0 +1,142 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { async, ComponentFixture, TestBed, tick, fakeAsync, flush } from '@angular/core/testing';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MockComponent } from 'ng2-mock-component';
+import { using } from 'app/testing/spec-helpers';
+
+import { ChefNotificationComponent } from './notification.component';
+import { Type } from 'app/entities/notifications/notification.model';
+
+describe('ChefNotificationComponent', () => {
+  let component: ChefNotificationComponent;
+  let fixture: ComponentFixture<ChefNotificationComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      imports: [ BrowserAnimationsModule ],
+      declarations: [
+        ChefNotificationComponent,
+        MockComponent({ selector: 'chef-icon' })
+      ],
+      schemas: [ CUSTOM_ELEMENTS_SCHEMA ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(ChefNotificationComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  const genNotification = (type: Type, timeout: number) => {
+    component.type = type;
+    component.timeout = timeout;
+  };
+
+  describe('generates a notification', () => {
+
+    it('and defaults to type of info and timeout of 8', () => {
+      expect(component.type).toBe('info');
+      expect(component.timeout).toEqual(8);
+    });
+
+    using([
+      [Type.info, 10],
+      [Type.info, 7],
+      [Type.info, 4],
+      [Type.info, 2],
+      [Type.info, 0],
+      [Type.error, 43],
+      [Type.error, 13],
+      [Type.error, 3],
+      [Type.error, 23]
+    ], function (type: Type, timeout: number) {
+          it('and reads type and timeout from inputs', () => {
+            genNotification(type, timeout);
+
+            expect(component.type).toBe(type);
+            expect(component.timeout).toEqual(timeout);
+          });
+        });
+  });
+
+  describe('ngAfterViewInit', () => {
+
+    using([
+      [Type.info, 10, false],
+      [Type.info, 0, true],
+      [Type.info, 4, false],
+      [Type.info, 2, false],
+      [Type.info, 0, true],
+      [Type.error, 43, false],
+      [Type.error, 0, true],
+      [Type.error, 7, false]
+    ], function (type: Type, timeout: number, isInfinite: boolean) {
+      it('determinites notification timeout is inifinite when timeout equals 0', () => {
+        genNotification(type, timeout);
+        component.ngAfterViewInit();
+
+        expect(component.isInfinite).toBe(isInfinite);
+      });
+    });
+
+    using([
+      [Type.info, 1, false],
+      [Type.info, 2, false],
+      [Type.error, 3, false]
+    ], function (type: Type, timeout: number, isInfinite: boolean) {
+      it('notification should close when infinite', fakeAsync(() => {
```

Comment:
This comment applies to all your data-driven tests here:
Remember that you need to make the test names unique. That simply means one or more parameters from each test case needs to be embedded in the test name string. Often, that involves adding another test parameter that I typically call `description` that describes what the test does.

(Why? When a test fails, it will say _"notification should close when infinite" failed_. So which of the three tests here failed? Second, some test runners generate a report of the tests run; seeing 8 occurrences of undifferentiated test "xyz" in the results list is not very useful.)